asr1k_neutron_l3/neutron/services/service_drivers/asr1k/fwaas_driver.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2021 SAP SE
+#
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+from neutron_fwaas.services.firewall.service_drivers import driver_api
+from neutron_lib import constants as nl_constants
+
+from oslo_log import helpers as log_helpers
+from oslo_log import log as logging
+
+from asr1k_neutron_l3.plugins.db import asr1k_db
+from asr1k_neutron_l3.plugins.l3.rpc import ask1k_l3_notifier
+
+LOG = logging.getLogger(__name__)
+
+
+class ASR1KFWaaSDriver(driver_api.FirewallDriverDB, driver_api.FirewallDriverRPCMixin):
```

Comment:
Why do we inherint from `FirewallDriverRPCMixin`? From the source it looks like all this does is a) provide a `start_rpc_listeners()` method and b) tries to invoke RPC workers in neutron-api (which looks like an upstream bug, which we also could correct). But as this is the driver we're running inside neutron ourselves and already have set up all the rpc we need I think we can drop the inheritance here.